st2common/st2common/services/workflows.py

```diff
@@ -388,12 +388,15 @@ def request_cancellation(ac_ex_db):
     root_ac_ex_db = ac_svc.get_root_execution(ac_ex_db)
 
     if root_ac_ex_db != ac_ex_db and root_ac_ex_db.status not in ac_const.LIVEACTION_CANCEL_STATES:
-        LOG.info('[%s] Cascading cancelation request to parent workflow.', wf_ac_ex_id)
+        LOG.info('[%s] Cascading cancellation request to parent workflow.', wf_ac_ex_id)
+        root_lv_ac_db = lv_db_access.LiveAction.get(id=root_ac_ex_db.liveaction['id'])
+        ac_svc.request_cancellation(root_lv_ac_db, None)
+    elif root_ac_ex_db == ac_ex_db and root_ac_ex_db.status == ac_const.LIVEACTION_STATUS_CANCELING:
```

Comment:
This is a good start, allowing users to cancel the workflow again even if it's in a canceling state.  Can you focus on this workaround first?  Currently, when this issue happens, the workflow and sub workflow is stuck in cancelling.  If we let users wait a bit and then allow them to run `st2 execution cancel` again, that would already improve the situation a bit.

Reply:
Actually revisiting this, I don't think adding the elif here for canceling will work.  This will just call the main workflow and ask it to cancel again.  The main workflow will then cascade the cancellation down back to this subworkflow.  And this may end up looping indefinitely.